actionview/lib/action_view/helpers/url_helper.rb

```diff
@@ -542,7 +542,7 @@ def current_page?(options, check_parameters: false)
 
         return false unless request.get? || request.head?
 
-        check_parameters ||= !options.is_a?(String) && options.try(:delete, :check_parameters)
+        check_parameters ||= options.respond_to?(:to_hash) && options.try(:delete, :check_parameters)
```

Comment:
Thanks for your input. Sounds good to me, just want to point out that if `current_page?` gets called with a `ActionController::Parameters` then the check_parameter thing won't work.